### ISSUE TITLE
Add more robust error resolution when adding plugin URL.

### ISSUE THREAD

**Issue by lilchurro** [CyberReboot/vent#438]
Steps to reproduce:
1. From main menu, choose Plugins --> Add new plugin --> [submit repo URL and personal user creds]
2. Note error... though, not sure if I'm providing unexpected input.
```Traceback (most recent call last):
  File "/home/alice/dev/bin/vent", line 4, in <module>
    __import__('pkg_resources').run_script('vent==0.3.2.dev0', 'vent')
  File "/home/alice/dev/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/alice/dev/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1511, in run_script
    exec(script_code, namespace, namespace)
  File "/home/alice/dev/local/lib/python2.7/site-packages/vent-0.3.2.dev0-py2.7.egg/EGG-INFO/scripts/vent", line 11, in <module>
    
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/apNPSApplication.py", line 30, in run
    return npyssafewrapper.wrapper(self.__remove_argument_call_main)
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/npyssafewrapper.py", line 41, in wrapper
    wrapper_no_fork(call_function)
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/npyssafewrapper.py", line 97, in wrapper_no_fork
    return_code = call_function(_SCREEN)    
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/apNPSApplication.py", line 25, in __remove_argument_call_main
    return self.main()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/apNPSApplicationManaged.py", line 172, in main
    self._THISFORM.edit()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/fmActionForm.py", line 66, in edit
    self._widgets__[self.editw].edit()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/wgwidget.py", line 458, in edit
    self._edit_loop()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/wgwidget.py", line 474, in _edit_loop
    self.get_and_use_key_press()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/wgwidget.py", line 575, in get_and_use_key_press
    return self.try_while_waiting()
  File "/home/alice/dev/local/lib/python2.7/site-packages/npyscreen/wgwidget.py", line 557, in try_while_waiting
    self.parent.while_waiting()
  File "build/bdist.linux-x86_64/egg/vent/menus/add_options.py", line 29, in while_waiting
  File "build/bdist.linux-x86_64/egg/vent/menus/add_options.py", line 17, in repo_values
  File "build/bdist.linux-x86_64/egg/vent/api/plugins.py", line 51, in repo_branches
  File "/usr/lib/python2.7/subprocess.py", line 574, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['git', 'pull', '--all']' returned non-zero exit status 128```

**Comment by cglewis** [comment]
oh interesting. do you have a repo for making this repeatable?  looks like there's a case where `git pull --all` returns 128 instead of 0 for your repo, which is unexpected and not handled.

**Comment by lilchurro** [comment]
I was just using the default cyberreboot/vent-plugins, but not sure what it's got an issue with. (Credentials? Something else?) It takes me to a menu with "Branch: " shortly before failing.

**Comment by cprafullchandra** [comment]
I can reproduce this if I put an invalid github repo URL in.

**Comment by cglewis** [comment]
this should be resolved with #450, if the problem persists, please reopen
